cow_instrument/InstrumentTree.cpp

```diff
@@ -76,44 +91,54 @@ V3D InstrumentTree::samplePos() const {
   return V3D{0, 0, 10}; // HACK!
 }
 
-size_t InstrumentTree::nDetectors() const { return m_detectorVec.size(); }
+size_t InstrumentTree::nDetectors() const { return m_detectorVec->size(); }
 
-std::unique_ptr<const InstrumentTree>
-InstrumentTree::modify(size_t node, const Command &command) const {
+bool InstrumentTree::modify(size_t nodeIndex, const Command &command) {
 
-  auto newNodes(m_nodes);
-  std::for_each(newNodes.begin(), newNodes.end(),
+  CowPtr<std::vector<Node>> newNodes(m_nodes);
+  std::for_each(newNodes->begin(), newNodes->end(),
                 [](Node &node) { node.incrementVersion(); });
```

Comment:
@OwenArnold My point is, why do you manually copy the node vector, doesn't COW do everything for you (i.e., you could avoid having `newNodes`)?